tests/unit/testplan/web_ui/test_web_app.py

```diff
@@ -77,108 +77,92 @@ def _create_tmp_data_files(base_dir):
-        tp_web_app.config["TESTPLAN_REPORT_NAME"] = "report.json"
-        tp_web_app.config["TESTING"] = True
-        tp_web_app.static_folder = os.path.abspath(
-            os.path.join(
-                tp_web_app.config["STATIC_PATH"], "testing", "build", "static"
-            )
-        )
-        self.app = tp_web_app
-        self.client = tp_web_app.test_client()
-
-    def teardown_method(self, _):
-        """Remove fake static files."""
-        shutil.rmtree(self.static_dir)
-
-    def test_testplan_index(self):
-        """Does /testplan/<uid> return the correct index.html page."""
-        # Use correct UID, expect 200 response.
-        response = self.client.get(
-            "/testplan/{}".format(STATIC_REPORTS["testing"]["uid"])
-        )
-        expected_contents = str(STATIC_REPORTS["testing"]["contents"])
-        assert response.status_code == 200
-        assert expected_contents in str(response.data)
-
-        # Use incorrect UID, still expect 200 response.
-        response = self.client.get("/testplan/123")
-        assert response.status_code == 200
-        assert expected_contents in str(response.data)
-
-    def test_testplan_static(self):
-        """
-        Does /testplan/static/static.file respond with the static.file file.
-        """
-        # Use correct UID, expect 200 response
-        response = self.client.get("/static/static.file")
-        expected_contents = str(STATIC_REPORTS["testing"]["contents"])
-        assert response.status_code == 200
-        assert expected_contents in str(response.data)
-
-        # This endpoint should return the index.html page no matter what UID is
-        # sent
-        response = self.client.get("/testplan/static/123")
-        assert response.status_code == 200
-        assert expected_contents in str(response.data)
-
-
-class TestDataEndpoints(object):
+    # Use correct UID, expect 200 response
+    response = webapp_test_client.get("/static/static.file")
+    expected_contents = str(STATIC_REPORTS["testing"]["contents"])
+    assert response.status_code == 200
+    assert expected_contents in str(response.data)
+
+    # This endpoint should return the index.html page no matter what UID is
+    # sent
+    response = webapp_test_client.get("/testplan/static/123")
+    assert response.status_code == 200
+    assert expected_contents in str(response.data)
+
+
+def test_testplan_report(webapp_test_client):
+    """Does /api/v1/reports/<uid>/report return the correct report.json file."""
+    # Use correct UID, expect 200 response.
+    path = "/api/v1/reports/{}".format(DATA_REPORTS["testplan"]["uid"])
+    response = webapp_test_client.get(path)
+    expected_contents = str(DATA_REPORTS["testplan"]["contents"])
+    assert response.status_code == 200
+    assert expected_contents in str(response.data)
+
+    # Use incorrect UID, still expect 200 response.
+    response = webapp_test_client.get("/api/v1/reports/123")
+    assert response.status_code == 200
```

Comment:
I thought we are returning a 404 here?

Reply:
This is the --ui flow here. This case there is only one report that just generated during the run, so the report id do not count too much.